### PR TITLE
Task/mov 595 big data test

### PR DESCRIPTION
**Pull request by muthenberg**
This adds the test cases for the whole lifecycle as well as one to simulate 10 serialization of 10 hours of capturing

### CODE REVIEW

File: DataCapturing.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -26,13 +26,17 @@
 		3D1D9A94224D4F420053B1B7 /* V2TestData.sqlite in Resources */ = {isa = PBXBuildFile; fileRef = 3D1D9A91224D4F420053B1B7 /* V2TestData.sqlite */; };
```
Comment (muthenberg) [inline DataCapturing.xcodeproj/project.pbxproj]: As usual just ignore this file. It is generated by Xcode.

File: DataCapturing.xcodeproj/xcshareddata/xcbaselines/3D0BDEA91FB1B951004EA25E.xcbaseline/0215D036-B6EF-4343-A287-D0A58E604DF7.plist [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (muthenberg) [inline DataCapturing.xcodeproj/xcshareddata/xcbaselines/3D0BDEA91FB1B951004EA25E.xcbaseline/0215D036-B6EF-4343-A287-D0A58E604DF7.plist]: This and the following file are generated by Xcode during testing. It stores performance values captured during performance testing. Future performance tests are compared against these values.

File: DataCapturing/DataCapturingService.swift [hunk 3]
```diff
@@ -69,22 +69,19 @@ public class DataCapturingService: NSObject {
     }()
 
     /**
-     An API to store, retrieve and update captured data to the local system until the App
+     The *CoreData* stack used to store, retrieve and update captured data to the local system until the App
      can transmit it to a server.
      */
-    let persistenceLayer: PersistenceLayer
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: Sharing one persistence layer between tracks leads to an ugly race condition. I removed that by requiring the creation of a new persistence layer on demand. To create such a persistence layer one requires a CoreDataManager, which therefore now is an attribute. Sharing a CoreDataManager between threads is possible.

File: DataCapturing/DataCapturingService.swift [hunk 4]
```diff
@@ -69,22 +69,19 @@ public class DataCapturingService: NSObject {
...
     /// An in memory storage for geo locations, before they are written to disk.
-    private var locationsCache = [GeoLocation]()
+    var locationsCache = [GeoLocation]()
 
     /// The background queue used to capture data.
-    private let capturingQueue = DispatchQueue.global(qos: .userInitiated)
-
-    /// Synchronizes read and write operations on the `locationsCache` and the `accelerationsCache`.
-    private let cacheSynchronizationQueue = DispatchQueue(label: "cacheSynchronization", attributes: .concurrent)
+    let capturingQueue = DispatchQueue.global(qos: .userInitiated)
```
Comment (muthenberg) [inline DataCapturing/DataCapturingService.swift]: Only one queue is used now. This avoids a deadlock between them both and frees some system resources.

File: DataCapturingTests/DataMigrationTest.swift [hunk 5]
```diff
@@ -357,8 +357,8 @@ extension NSManagedObjectContext {
      */
     func destroyStore() {
         persistentStoreCoordinator?.persistentStores.forEach {
-            try? persistentStoreCoordinator?.remove($0)
-            try? persistentStoreCoordinator?.destroyPersistentStore(at: $0.url!, ofType: $0.type, options: nil)
+            ((try? persistentStoreCoordinator?.remove($0)) as ()??)
+            ((try? persistentStoreCoordinator?.destroyPersistentStore(at: $0.url!, ofType: $0.type, options: nil)) as ()??)
```
Comment (muthenberg) [inline DataCapturingTests/DataMigrationTest.swift]: I missed that stuff in the Swift 5 migration branch. Sorry. Adding it here now.

File: DataCapturing.xcodeproj/xcshareddata/xcbaselines/3D0BDEA91FB1B951004EA25E.xcbaseline/0215D036-B6EF-4343-A287-D0A58E604DF7.plist (hunk 2 above)
Comment (hb0) [reply DataCapturing.xcodeproj/xcshareddata/xcbaselines/3D0BDEA91FB1B951004EA25E.xcbaseline/0215D036-B6EF-4343-A287-D0A58E604DF7.plist]: That's cool. (monitoring the code performance)

**Pull request merged**